relayer/src/chain/cosmos.rs

```diff
@@ -379,12 +398,15 @@ impl CosmosSdkChain {
     fn send_tx_simulate(&self, tx: Tx) -> Result<SimulateResponse, Error> {
         crate::time!("tx simulate");
 
-        // The `tx` field of `SimulateRequest` was deprecated
-        // in favor of `tx_bytes`
+        // The `tx` field of `SimulateRequest` was deprecated in Cosmos SDK 0.43 in favor of `tx_bytes`.
         let mut tx_bytes = vec![];
         prost::Message::encode(&tx, &mut tx_bytes).unwrap();
+
         #[allow(deprecated)]
-        let req = SimulateRequest { tx: None, tx_bytes };
+        let req = SimulateRequest {
+            tx: Some(tx), // needed for simulation to go through with Cosmos SDK <  0.43
+            tx_bytes,     // needed for simulation to go through with Cosmos SDk >= 0.43
+        };
```

Comment:
ah, I guess this was the problem. Were you able to test the fix with both `< 0.43` and `>= 0.43`?

Reply:
Yes, I tested with both Gaia 5.0.5 and Gaia 6.0.0-rc1